src/subscribe.rs

```diff
@@ -0,0 +1,429 @@
+                        );
+                }
+            }
+            None => {
+                data.as_object_mut()
+                    .expect("data is always a json object")
+                    .insert(
+                        "subscription".to_string(),
+                        serde_json::json!({
+                            "sources": [source],
+                            "topics": [],
+                        }),
+                    );
+            }
+        }
+    }
+
+    match data
+        .as_object_mut()
+        .expect("data is always a json object")
+        .get_mut("subscription")
+    {
+        Some(sub) => {
+            let old_value = sub
+                .as_object_mut()
+                .expect("subscription is always a json object")
+                .insert("subscribed".to_string(), serde_json::Value::Bool(true));
+
+            if old_value.is_none() {
+                subscribed = true;
+            }
+
+            if let Some(name) = &subscriber.name {
+                sub.as_object_mut()
+                    .expect("subscription is always a json object")
+                    .insert(
+                        "name".to_string(),
+                        serde_json::Value::String(name.to_string()),
+                    );
+            }
+
+            if let Some(phone) = &subscriber.phone {
+                sub.as_object_mut()
+                    .expect("subscription is always a json object")
+                    .insert(
+                        "phone".to_string(),
+                        serde_json::Value::String(phone.to_string()),
+                    );
+            }
+
+            sub.as_object_mut()
+                .expect("subscription is always a json object")
+                .insert(
+                    "email".to_string(),
+                    serde_json::Value::String(subscriber.email.clone()),
+                );
+        }
+        None => {
+            data.as_object_mut()
+                .expect("data is always a json object")
+                .insert(
+                    "subscription".to_string(),
+                    serde_json::json!({
+                        "subscribed": true,
+                        "topics": [],
+                        "sources": [],
+                        "name": subscriber.name,
+                        "phone": subscriber.phone,
+                        "email": subscriber.email,
+                        "double_optin": false,
+                    }),
+                );
+
+            subscribed = true;
+        }
+    }
+
+    (subscribed, data)
+}
+
+pub fn email_from_address_from_env() -> (String, String) {
+    let email = ft_sdk::env::var("FASTN_SMTP_SENDER_EMAIL".to_string())
+        .unwrap_or_else(|| "support@fifthtry.com".to_string());
+    let name = ft_sdk::env::var("FASTN_SMTP_SENDER_NAME".to_string())
+        .unwrap_or_else(|| "FifthTry Team".to_string());
+
+    (name, email)
+}
```

Comment:
Let's move it to `ft-sdk`. Since it is used by `auth` also.

Reply:
Resolving this for now. This function will be removed in future when https://github.com/FifthTry/ft-sdk/pull/48 is merged :+1: